### DISCUSSION TITLE
same backup to multiple destinations (onsite and offsite)

### DISCUSSION THREAD

**Discussion by rhatguy** [djmaze/resticker#148] [General]
I have a similar question to the multiple sources discussion previously, but about multiple destinations.  I like to create onsite and offsite backups.  Is there a way to have the container just do the same backup twice, or to copy the backup from one repository to another or would I need to spin up a second instance of resticker for that?

**Comment by djmaze** [comment]
There is not integrated way of doing that. You could spin up a second backup container with the same configuration, but with a different repository. But I guess that's not optimal, performance-wise.

As `rclone` is part of the container, you could run it in the `POST_COMMANDS_SUCCESS` script in order to sync the data from the first repository to the second one. But that might be really slow (because you are then doing a remote->remote copy). Not sure what's the best solution here.

**Comment by rhatguy** [comment]
I wonder if a 4th container could be implemented easily (like backup, check, and prune today) that would run a restic copy operation from one repository to another.  I recently used this functionality to copy from v1 repositories to v2 repositories to take advantage of compression.  I could always setup a job external to the containers, but it seems that this would be to far of a stretch given the existing architecture of the app.  If one of the repositories is local lan based, the added network traffic is to much of a concern.

https://restic.readthedocs.io/en/latest/045_working_with_repos.html#copying-snapshots-between-repositories

**Reply by djmaze** [reply]
Mhh.. I could imagine we introduce such a container. Should not be too hard to implement. Personally I probably don't have time to implement this soon, but I would be open for pr's!
